handlers/handlers.go

```diff
@@ -395,121 +395,134 @@ func readPreviousReleases(w http.ResponseWriter, req *http.Request, cfg *config.
+			cancel()
 			return
 		}
 	}()
 
-	var options searchSDK.Options
-
-	options.Query = searchQuery
-
-	options.Headers = http.Header{
-		searchSDK.FlorenceToken: {Bearer + accessToken},
-		searchSDK.CollectionID:  {collectionID},
-	}
-
 	go func() {
 		defer wg.Done()
-
-		searchResp, respErr = searchC.GetSearch(ctx, options)
-		if respErr != nil {
-			log.Error(ctx, "getting search response from client failed for aggregation", respErr)
-			cancel()
+		// get cached navigation data
+		navigationCache, nErr = cacheList.Navigation.GetNavigationData(ctx, lang)
+		if nErr != nil {
+			log.Error(ctx, "failed to get navigation cache for aggregation", nErr)
 			return
 		}
 	}()
 
 	go func() {
 		defer wg.Done()
-
+		// get breadcrumbs
 		bc, bcErr = zc.GetBreadcrumb(ctx, accessToken, collectionID, lang, latestContentURL)
 		if bcErr != nil {
-			bc = []zebedeeCli.Breadcrumb{}
+			log.Warn(ctx, "getting breadcrumb response from client failed", log.FormatErrors([]error{bcErr}))
+			return
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		// get homepage content
+		homepageResp, homeErr = zc.GetHomepageContent(ctx, accessToken, collectionID, lang, homepagePath)
+		if homeErr != nil {
+			log.Warn(ctx, "unable to get homepage content", log.FormatErrors([]error{homeErr}))
 			return
 		}
 	}()
 
 	wg.Wait()
-	if respErr != nil || countErr != nil {
-		setStatusCode(w, req, respErr)
+	// handle errors
+	// 404 errors
+	if pdErr != nil || ptErr != nil {
+		w.WriteHeader(http.StatusNotFound)
 		return
 	}
 
-	basePage := rend.NewBasePageModel()
-	err = validateCurrentPage(ctx, cfg, validatedQueryParams, searchResp.Count)
-	if err != nil {
+	// 500 errors
+	if nErr != nil {
+		setStatusCode(w, req, nErr)
+		return
+	}
+
+	sanitisedParams := sanitiseQueryParams(allowedPreviousReleasesQueryParams, urlQuery)
+
+	validatedQueryParams, validationErrs := data.ReviewPreviousReleasesQueryWithParams(ctx, cfg, sanitisedParams, urlPath)
+	if len(validationErrs) > 0 {
+		log.Info(ctx, "validation of parameters failed for aggregation", log.Data{
+			"parameters": validationErrs,
+		})
+		// Errors are now mapped to the page model to output feedback to the user rather than
+		// a blank 400 error response.
+		m := mapper.CreatePreviousReleasesPage(cfg, req, rend.NewBasePageModel(), validatedQueryParams, &searchModels.SearchResponse{}, lang, homepageResp, "", navigationCache, template, cache.Topic{}, validationErrs, pageData, bc)
+		rend.BuildPage(w, m, template)
+		return
+	}
+
+	searchQuery := data.SetParentTypeOnSearchAPIQuery(validatedQueryParams, pageData.Type)
+
+	var (
+		respErr error
+
+		options searchSDK.Options
+	)
+
+	options.Query = searchQuery
+
+	options.Headers = http.Header{
+		searchSDK.FlorenceToken: {Bearer + accessToken},
+		searchSDK.CollectionID:  {collectionID},
+	}
+
+	searchResp, respErr := searchC.GetSearch(ctx, options)
+	if respErr != nil {
+		log.Error(ctx, "getting search response from client failed for aggregation", respErr)
```

Comment:
Not aggregation page.

Reply:
🤦‍♂️ - foiled by own copying and pasting